File: app/src/main/java/com/pluu/event/RxBus.java
package com.pluu.event;

import io.reactivex.Observable;
import io.reactivex.subjects.PublishSubject;

/**
 * RxBus from RxJava
 * Created by PLUUSYSTEM-NEW on 2016-03-21.
 */
public class RxBus {
    private final PublishSubject<Object> mBus = PublishSubject.create();

    public RxBus() { }

    public void send(Object o) {
        mBus.onNext(o);
    }

    public Observable<Object> toObservable() {
        return mBus;
    }

    public boolean hasObservers() {
        return mBus.hasObservers();
    }
}

File: app/src/main/java/com/pluu/event/RxBus.kt
package com.pluu.event

import io.reactivex.Observable
import io.reactivex.subjects.PublishSubject

/**
 * RxBus from RxJava
 * Created by pluu on 2017-04-19.
 */
class RxBus {
    private val mBus = PublishSubject.create<Any>()

    fun send(o: Any) {
        mBus.onNext(o)
    }

    fun toObservable(): Observable<Any> {
        return mBus
    }

    fun hasObservers(): Boolean {
        return mBus.hasObservers()
    }
}
